source/common/upstream/cluster_manager_impl.h

```diff
@@ -259,8 +262,8 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
 
   Config::SubscriptionFactory& subscriptionFactory() override { return subscription_factory_; }
 
-  void
-  initializeSecondaryClusters(const envoy::config::bootstrap::v3::Bootstrap& bootstrap) override;
+  void initializeSecondaryClusters(const envoy::config::bootstrap::v3::Bootstrap& bootstrap,
+                                   Stats::StoreRootPtr& load_report_stats_store) override;
```

Comment:
I'm wondering if we could have the load reporter look up the relevant stat (as string) and build out a stats store locally.

Reply:
Looking up a stat as a string is more complex than it would appear. See https://github.com/envoyproxy/envoy/issues/11799 .

We should be able to do it but it will be very slow and you'll want to catche it.

Reply:
Yeah, it should only be necessary to do this once for any activate cluster in the load reporter.

Reply:
If we want to use the original store root, I think we'll have to come up with some kind of scheme where the main histogram merge for flushing metrics will have to skip these metrics. @mahmoudas and I had considered adding a `secondaryHistograms` that aren't flushed, but that seemed hacky.

Another issue was that `IsolatedStoreImpl`'s histogram implementation didn't seem like it would be performant enough to use in `routing.cc`